// ng-lightning/esm2015/lib/files/file.js
import { Component, Input, ChangeDetectionStrategy, ElementRef, Renderer2, HostBinding } from '@angular/core';
export class NglFile {
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.renderer.addClass(this.element.nativeElement, 'slds-file');
        this.renderer.addClass(this.element.nativeElement, 'slds-file_card');
    }
}
NglFile.decorators = [
    { type: Component, args: [{
                selector: 'ngl-file',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: "\n<figure>\n  <ng-content></ng-content>\n  <figcaption class=\"slds-file__title slds-file__title_card\" *ngIf=\"text\">\n    <div class=\"slds-media slds-media_small slds-media_center\">\n      <div class=\"slds-media__figure slds-line-height_reset\" *ngIf=\"iconName\">\n        <ngl-icon [iconName]=\"iconName\"></ngl-icon>\n      </div>\n      <div class=\"slds-media__body\"><span class=\"slds-file__text slds-truncate\" [title]=\"text\" [nglInternalOutlet]=\"text\"></span></div>\n    </div>\n  </figcaption>\n</figure>"
            },] }
];
NglFile.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NglFile.propDecorators = {
    text: [{ type: HostBinding, args: ['class.slds-has-title',] }, { type: Input }],
    iconName: [{ type: Input }]
};
