// ng-lightning/esm2015/lib/accordion/accordion.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter, ChangeDetectionStrategy, ElementRef, Renderer2, ContentChildren } from '@angular/core';
import { NglAccordionSection } from './accordion-section';
import { isOptionSelected, addOptionToSelection } from '../util/util';
import { InputBoolean } from '../util/convert';
export class NglAccordion {
    constructor(element, renderer) {
        this.activeNameChange = new EventEmitter();
        /**
         * Whether we allow multiple sections open at a time.
         */
        this.multiple = false;
        renderer.addClass(element.nativeElement, 'slds-accordion');
    }
    toggle(section) {
        const active = addOptionToSelection(section.name, this.activeName, this.multiple, true);
        this.activeNameChange.emit(active);
    }
    isActive(section) {
        return isOptionSelected(section.name, this.activeName, this.multiple);
    }
}
NglAccordion.decorators = [
    { type: Component, args: [{
                selector: 'ngl-accordion,[ngl-accordion]',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: "\n<li *ngFor=\"let section of sections\" nglAccordionItem [isActive]=\"isActive(section)\" [section]=\"section\" (toggle)=\"toggle(section)\"></li>"
            },] }
];
NglAccordion.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NglAccordion.propDecorators = {
    activeName: [{ type: Input }],
    activeNameChange: [{ type: Output }],
    multiple: [{ type: Input }],
    sections: [{ type: ContentChildren, args: [NglAccordionSection,] }]
};
__decorate([
    InputBoolean()
], NglAccordion.prototype, "multiple", void 0);
