// ng-lightning/esm2015/lib/colorpicker/swatches/trigger.js
import { Directive, ElementRef, Renderer2, Input, Output, EventEmitter, HostListener, HostBinding } from '@angular/core';
export class NglColorpickerSwatchTrigger {
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.selectedChange = new EventEmitter();
        this.renderer.addClass(this.el.nativeElement, 'slds-color-picker__swatch-trigger');
        this.renderer.setAttribute(this.el.nativeElement, 'role', 'option');
    }
    onSelect() {
        return this.selectedChange.emit();
    }
    focus() {
        this.el.nativeElement.focus();
        this.onSelect();
    }
}
NglColorpickerSwatchTrigger.decorators = [
    { type: Directive, args: [{
                selector: '[nglColorpickerSwatchTrigger]',
            },] }
];
NglColorpickerSwatchTrigger.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NglColorpickerSwatchTrigger.propDecorators = {
    selected: [{ type: HostBinding, args: ['class.ngl-color-picker__swatch-selected',] }, { type: Input }],
    selectedChange: [{ type: Output }],
    onSelect: [{ type: HostListener, args: ['click',] }]
};
