// ng-lightning/esm2015/lib/checkboxes/button/checkbox-button.js
import { Component, Input, ChangeDetectionStrategy, ContentChild, ChangeDetectorRef } from '@angular/core';
import { NglCheckboxInput } from '../input/input';
export class NglCheckboxButton {
    constructor(cd) {
        this.cd = cd;
    }
    ngAfterContentInit() {
        if (!this.input) {
            throw Error(`[ng-lightning] Couldn't find an <input type="checkbox"> with [ngl] attribute inside ${this}`);
        }
        this._uid = this.input.id;
        this.cd.detectChanges();
        this.input.addClass('slds-assistive-text');
    }
}
NglCheckboxButton.decorators = [
    { type: Component, args: [{
                selector: 'ngl-checkbox-button',
                template: "\n<ng-content></ng-content>\n<label class=\"slds-checkbox_faux\" [attr.for]=\"_uid\"><span class=\"slds-assistive-text\" [nglInternalOutlet]=\"label\"></span></label>",
                changeDetection: ChangeDetectionStrategy.OnPush,
                host: {
                    '[class.slds-checkbox_add-button]': 'true',
                }
            },] }
];
NglCheckboxButton.ctorParameters = () => [
    { type: ChangeDetectorRef }
];
NglCheckboxButton.propDecorators = {
    input: [{ type: ContentChild, args: [NglCheckboxInput, { static: true },] }],
    label: [{ type: Input }]
};
