// ng-lightning/esm2015/lib/buttons/button-stateful.js
import { __decorate } from "tslib";
import { Directive, Input, Output, EventEmitter, HostListener, ElementRef, Renderer2 } from '@angular/core';
import { InputBoolean } from '../util/convert';
import { HostService } from '../common/host/host.service';
export class NglButtonStateful {
    constructor(el, renderer, hostService) {
        this.el = el;
        this.renderer = renderer;
        this.hostService = hostService;
        /**
         * Triggered when the button is clicked.
         */
        this.stateChange = new EventEmitter();
        /**
         * Appearance.
         */
        this.variant = 'neutral';
        this.renderer.addClass(this.el.nativeElement, 'slds-button');
        this.renderer.addClass(this.el.nativeElement, 'slds-button_stateful');
        this.renderer.setAttribute(this.el.nativeElement, 'aria-live', 'assertive');
    }
    onSelectChange() {
        this.stateChange.emit(!this.state);
    }
    onFocusToggle(focused) {
        this.focused = !!+focused;
        if (!this.focused) {
            this.setHostClass();
        }
    }
    ngOnInit() {
        this.setHostClass();
    }
    ngOnChanges() {
        this.setHostClass();
    }
    setHostClass() {
        this.hostService.updateClass(this.el, {
            [`slds-button_${this.variant === 'text' ? 'reset' : this.variant}`]: !!this.variant,
            [`slds-is-selected-clicked`]: this.state && this.focused,
            [`slds-is-selected`]: this.state && !this.focused,
            [`slds-not-selected`]: !this.state,
        });
    }
}
NglButtonStateful.decorators = [
    { type: Directive, args: [{
                selector: '[nglButtonStateful]',
                providers: [HostService],
            },] }
];
NglButtonStateful.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 },
    { type: HostService }
];
NglButtonStateful.propDecorators = {
    state: [{ type: Input }],
    stateChange: [{ type: Output }],
    variant: [{ type: Input }],
    onSelectChange: [{ type: HostListener, args: ['click',] }],
    onFocusToggle: [{ type: HostListener, args: ['focus', ['1'],] }, { type: HostListener, args: ['blur', ['0'],] }]
};
__decorate([
    InputBoolean()
], NglButtonStateful.prototype, "state", void 0);
