// ng-lightning/esm2015/lib/textarea/input/input.js
import { Directive, HostBinding, ElementRef, Input, Renderer2 } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { toBoolean } from '../../util/convert';
import { uniqueId } from '../../util/util';
export class NglTextareaInput {
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.ɵRequiredSubject = new BehaviorSubject(false);
        if (!this.el.nativeElement.id) {
            this.renderer.setAttribute(this.el.nativeElement, 'id', uniqueId('textarea'));
        }
    }
    set required(required) {
        this.ɵRequiredSubject.next(toBoolean(required));
    }
    get id() {
        return this.el.nativeElement.id;
    }
}
NglTextareaInput.decorators = [
    { type: Directive, args: [{
                selector: 'textarea[ngl]',
                host: {
                    '[class.slds-textarea]': 'true',
                },
            },] }
];
NglTextareaInput.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NglTextareaInput.propDecorators = {
    describedBy: [{ type: HostBinding, args: ['attr.aria-describedby',] }],
    required: [{ type: Input }]
};
