// ng-lightning/esm2015/lib/colorpicker/custom/colorpicker-custom.js
import { Component, ElementRef, Renderer2, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';
import { getHexFromHsv, getHsvFromHex } from '../util';
export class NglColorpickerCustom {
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.hsvChange = new EventEmitter();
        this.renderer.addClass(this.el.nativeElement, 'slds-color-picker__custom');
    }
    ngOnChanges(changes) {
        if (changes.hsv) {
            this.hex = getHexFromHsv(this.hsv);
        }
    }
    onHsvChange($event) {
        this.hsvChange.emit($event);
    }
    onHexChange(hex) {
        const hsv = getHsvFromHex(hex);
        this.hsvChange.emit(hsv);
    }
}
NglColorpickerCustom.decorators = [
    { type: Component, args: [{
                selector: 'ngl-colorpicker-custom',
                template: "\n<ngl-colorpicker-range [hsv]=\"hsv\" (hsvChange)=\"onHsvChange($event)\"></ngl-colorpicker-range>\n<ngl-colorpicker-inputs [hex]=\"hex\" (hexChange)=\"onHexChange($event)\"></ngl-colorpicker-inputs>",
                changeDetection: ChangeDetectionStrategy.OnPush
            },] }
];
NglColorpickerCustom.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NglColorpickerCustom.propDecorators = {
    hsv: [{ type: Input }],
    hsvChange: [{ type: Output }]
};
