// ng-lightning/esm2015/lib/forms/label.js
import { __decorate } from "tslib";
import { Component, Input, ChangeDetectionStrategy, ElementRef, Renderer2 } from '@angular/core';
import { InputBoolean } from '../util/convert';
export class NglFormLabel {
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.klass = 'slds-form-element__label';
    }
    ngOnInit() {
        this.renderer.addClass(this.element.nativeElement, this.klass);
    }
}
NglFormLabel.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'label[nglFormLabel]',
                template: "<abbr class=\"slds-required\" *ngIf=\"required\" title=\"Required\">*</abbr><span [nglInternalOutlet]=\"label\"></span>\n<ng-content></ng-content>",
                changeDetection: ChangeDetectionStrategy.OnPush
            },] }
];
NglFormLabel.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NglFormLabel.propDecorators = {
    label: [{ type: Input, args: ['nglFormLabel',] }],
    klass: [{ type: Input, args: ['nglFormLabelClass',] }],
    required: [{ type: Input }]
};
__decorate([
    InputBoolean()
], NglFormLabel.prototype, "required", void 0);
