// ng-lightning/esm2015/lib/datatables/cell-internal.js
import { Component, ChangeDetectionStrategy, Input, HostBinding } from '@angular/core';
export class NglInternalDatatableCell {
    get dataLabel() {
        return this.column.heading;
    }
    ngOnChanges() {
        this.context = {
            $implicit: this.value,
            row: this.row,
            index: this.index,
        };
    }
    get value() {
        const { key } = this.column;
        return key ? this.row[key] : null;
    }
}
NglInternalDatatableCell.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'td[nglDatatatableCell_]',
                template: "\n<div [class.slds-truncate]=\"column.truncate\" [attr.title]=\"column.truncate ? value : null\">\n  <ng-container *ngIf=\"column.cellTpl; else stringTpl\" [ngTemplateOutlet]=\"column.cellTpl.templateRef\" [ngTemplateOutletContext]=\"context\"></ng-container>\n  <ng-template #stringTpl>{{ value }}</ng-template>\n</div>",
                changeDetection: ChangeDetectionStrategy.OnPush
            },] }
];
NglInternalDatatableCell.propDecorators = {
    row: [{ type: Input }],
    column: [{ type: Input }],
    index: [{ type: Input }],
    dataLabel: [{ type: HostBinding, args: ['attr.data-label',] }]
};
