// ng-lightning/esm2015/lib/datatables/column.js
import { __decorate } from "tslib";
import { Directive, Input, ContentChild } from '@angular/core';
import { NglDatatableCell } from './cell';
import { NglDatatableHeadingTemplate } from './heading';
import { InputBoolean } from '../util/convert';
export class NglDatatableColumn {
    constructor() {
        this.sortable = false;
        this.truncate = false;
    }
}
NglDatatableColumn.decorators = [
    { type: Directive, args: [{
                // tslint:disable-next-line:directive-selector
                selector: 'ngl-datatable-column',
            },] }
];
NglDatatableColumn.propDecorators = {
    heading: [{ type: Input }],
    key: [{ type: Input }],
    headClass: [{ type: Input }],
    cellClass: [{ type: Input }],
    sortable: [{ type: Input }],
    truncate: [{ type: Input }],
    cellTpl: [{ type: ContentChild, args: [NglDatatableCell,] }],
    headingTpl: [{ type: ContentChild, args: [NglDatatableHeadingTemplate,] }]
};
__decorate([
    InputBoolean()
], NglDatatableColumn.prototype, "sortable", void 0);
__decorate([
    InputBoolean()
], NglDatatableColumn.prototype, "truncate", void 0);
