// ng-lightning/esm2015/lib/carousel/carousel-indicator.js
import { __decorate } from "tslib";
import { Directive, HostBinding, Input, ElementRef } from '@angular/core';
import { InputBoolean } from '../util/convert';
import { uniqueId } from '../util/util';
export class NglCarouselIndicator {
    constructor(el) {
        this.el = el;
        this.uid = uniqueId('carousel-indicator');
    }
    get tabindex() {
        return this.isActive ? 0 : -1;
    }
    ngOnChanges(changes) {
        this.image.active = this.isActive;
        if (changes.image) {
            this.image.labelledby = this.uid;
        }
    }
    focus() {
        this.el.nativeElement.focus();
    }
}
NglCarouselIndicator.decorators = [
    { type: Directive, args: [{
                selector: '[nglCarouselIndicator]',
            },] }
];
NglCarouselIndicator.ctorParameters = () => [
    { type: ElementRef }
];
NglCarouselIndicator.propDecorators = {
    isActive: [{ type: HostBinding, args: ['class.slds-is-active',] }, { type: HostBinding, args: ['attr.aria-selected',] }, { type: Input }],
    tabindex: [{ type: HostBinding, args: ['attr.tabindex',] }],
    image: [{ type: Input }],
    uid: [{ type: HostBinding, args: ['attr.id',] }]
};
__decorate([
    InputBoolean()
], NglCarouselIndicator.prototype, "isActive", void 0);
